mockfaster/src/main/java/com/decathlon/tzatziki/utils/MockFaster.java

```diff
@@ -178,16 +192,23 @@ public static Integer localPort() {
     private static final Set<Pattern> PATH_PATTERNS = new LinkedHashSet<>();
 
     public static List<LogEventRequestAndResponse> retrieveRequestResponses(HttpRequest httpRequest) {
-        PATH_PATTERNS.stream()
-                .map(pathPattern -> pathPattern.matcher(httpRequest.getPath().getValue()))
-                .filter(Matcher::matches)
-                .filter(matcher -> matcher.groupCount() > 0)
-                .findFirst()
-                .ifPresent(matcher -> {
+        List<Pattern> patternArrayList = new ArrayList<>(PATH_PATTERNS);
+        Collections.reverse(patternArrayList);
+        for (Pattern pattern : patternArrayList) {
+            if (pattern.pattern().equals(httpRequest.getPath().getValue())) {
+                break;
+            } else {
+                Matcher matcher = pattern.matcher(httpRequest.getPath().getValue());
+                if (matcher.matches() && matcher.groupCount() > 0) {
```

Comment:
I would say to break upon matches in any case (even if groupCount == 0) and do the loop only if groups are there

Reply:
Fixed ! Thanks for the suggestion !